Modules/Module_Processing/xASL_init_SubjectList.m

```diff
@@ -74,7 +74,7 @@
         elseif ~strcmp(visitName(1:4), 'ses-')
             warning(['Illegal BIDS session definition for ' subjectName '_' visitName]);
         else
-            visitName = visitName(5:end); % if we find a BIDS session, we add it as _n visit suffix in ExploreASL legacy format
+            visitName = xASL_adm_CorrectName(visitName(5:end), 2); % if we find a BIDS session, we add it as visit suffix in ExploreASL legacy format
```

Comment:
No the BIDS session should already be corrected. I would not alter the visit/session names as they should in ExploreASL be identical to the names in BIDS.

So I would rather issue an error if there are illegal symbols instead of removing them without notifying the user.

Image use case 1:

sourcedata/participant1_2024+02+01

-> this is easy, it will just remove the illegal symbols

but use case 2, where we are not the ones that created rawdata:

rawdata/sub-001/ses-2024\02\01

Perhaps we should still correct this automatically, but then we should have at least a detailed warning

Reply:
See the relevant discussion inside the issue.

1. We remove illegal characters in visit names when doing DICOM to BIDS. 
2. We keep visit names when doing BIDS to Legacy as illegal names were already removed in DICOM to BIDS
3. But we need to check visit names when doing BIDS to Legacy when we have external data.

So I would keep automatic correct - but a very good point that we should give a warning for every such case. Agree?

Reply:
1. Agreed, but we add a boolean to xASL_adm_CorrectName for issuing a warning if we changed something, that defaults to false. And in this case it is set to true, so if correcting the visit name the user will be warned.
2. Agreed
3. Not agreed. A pipeline that has BIDS input, assumes BIDS. Something that is nearly BIDS is not BIDS, so will break the pipeline. So A) do you mean that BIDS supports illegal characters (I would not think so, but I don't know?) B) if you want to add failsafes, that is fine with me, but it should be done at the point of BIDS2Legacy. You inserted this at the list loading. We cannot change things at the point of list loading, it has to be done at the point of creating legacy derivatives.

E.g.,

1. bids.layout reads rawdata in xASL_init_SubjectList

rawdata/sub-001/ses-2024\02\01 

becomes visit 2024\02\01 which is fine as string

2. then when we convert BIDS2Legacy, we do the conversion, both internally (visit name) and on the disk (newly saved derivatives/ExploreASL/sub-001_20240201 )

Reply:
1. xASL_adm_CorrectName doesn't accept a boolean for reporting. We report changes ourselves after applying xASL_adm_CorrectName.
2. OK
3. We accept BIDS. But A) BIDS doesn't forbid special characters by default, so we need to do it ourselves. B) We of course do the failsafe on both places. We have to do it also already when loading here, because we save the output to x-struct, so we can't have the non-corrected name there.

Reply:
1. xASL_adm_CorrectName should get a boolean for reporting, is what I say. This is much easier than having to code each instance "if changed, then report". And it serves as a reminder to provide feedback to the user for changing the name.
3. We have to have the non-corrected name here. I don't know what you mean by x-struct, but this is the point where a population-wide subject-list is read. Has nothing to do with the subject-specific x-struct.

A. When reading subject-list to memory: nothing changes in any of the subject/session/run name
B. When converting BIDS to legacy per subject, we correct the name in memory and for what we write to the disk.

BTW: assuming that rawdata and derivatives/ExploreASL always are on the same disk/filesystem etc, how can something be OK for the rawdata file/folder but not when saving as derivatives? It is exactly the same disk/filesystem with the same limitations/illegal characters. In Matlab strings, we can have whatever we want, so we're not restricted by our internal memory. So is this part even needed?

Reply:
1. This doesn't make sense as each adm_Correct name is applied to a very different task. So one fixes subject names, one visit names, one path name. So having reporting inside would report a name change, but it would be very unclear what kind of change that was.
2. It doesn't load it to memory. It saves it to x.dataset.TotalSubjects, which I assume is used later, so it needs to corrected name. Anyway - that's a useless comment, I keep both the corrected (saved in x-struct and used for writing to legacy) and uncorrected name (to read rawdata) and things work.
3. We have agreed upon removing non-alphanumeric characters from BIDS to Legacy. It's all written in the issue. You should have said you didn't like it before, and not change the plan in the middle of implementation.

Reply:
We added an output option to xASL_adm_CorrectName called bCorrected that is set to TRUE when some modifs were done. It actually makes no sense to return the original as the original was the input. And the new name is the output.

1. We use the output boolean of adm_CorrectName to decide on reporting the name changes.
2. I have removed the functionality of correcting names that are already in BIDS.